### ISSUE TITLE
2018 idioms: incorrect suggestion for root module and crate of same name

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang/rust#55759]
First reported at https://github.com/rust-lang/cargo/issues/6273

this code:

```rust
#![warn(rust_2018_idioms)]

extern crate time as std_time;

pub mod time {
    pub fn f() {
        println!("{:?}", crate::std_time::now());
    }
}

fn main() {}
```

[when compiled](https://play.rust-lang.org/?version=beta&mode=debug&edition=2018&gist=b3d0b4244408b208f6dc9ad9b60f7fa5) gives a warning that, when applied, produces code that doesn't compile

**Comment by alexcrichton** [comment]
@ehuss notes that [this code](https://github.com/rust-lang/rust/blame/15d770400eed9018f18bddf83dd65cb7789280a5/src/librustc_typeck/check_unused.rs#L184-L187) is likely relevant here.

**Comment by estebank** [comment]
Beta no longer emits this warning.

**Comment by ehuss** [comment]
Yea, appears to be fixed by #60252.  Although the test in that one is very similar, this issue is specific to a `use` ambiguity error.
